### ISSUE TITLE
Variables in pandas.DataFrame.query method are not recognized

### ISSUE THREAD

**Issue by meetashok** [PyCQA/flake8#1315]
The below code raises: `F841 local variable 'name' is assigned to but never used` even though the variable is being used in the `df.query` method

```
data = {
    "name": ["john", "jane", "jessica", "james"], 
    "age": [20, 10, 12, 15], 
}

df = pd.DataFrame(data)
name = "john"
df.query("name==@name")
```

**Comment by asottile** [comment]
**in the future, please fill out issue template or your issue will be closed without comment**

___

yeah, `eval` / `exec` hacks like `df.query` are intentionally not supported

here's a better way to write that code without involving frame hackery:

```python
df.query("name==@name", local_dict={'name': 'john'}) 
```

**Comment by asottile** [comment]
also note that as the issue template says:

```
*************************************************************************
NOTE: flake8 is a linting framework and does not implement any checks

if you are reporting a problem with a particular check, please track down
the plugin which implements that check.

some common ones:
- F###: https://github.com/pycqa/pyflakes
- E###, W###: https://github.com/pycqa/pycodestyle
*************************************************************************
```

**Comment by shortorian** [comment]
if, like me, you're dealing with a ton of similar queries across a large code base, you could also consider using helper functions like this
```python
def column_equality(a, b):
    return {'expr': '{} == @q'.format(a), 'local_dict': {'q': b}}

def column_isin(a, b):
    return {'expr': '{}.isin(@q)'.format(a), 'local_dict': {'q': b}}
```
which you can use this way
```
import pandas as pd
df = pd.DataFrame({'a': [1, 2], 'b': [10, 20]})

query_value = 20
print(df.query(**column_equality('b', query_value)))

query_iterable = [20]
print(df.query(**column_isin('b', query_iterable)))
```

**Comment by asottile** [comment]
doing formatting on strings which are about to be eval'd seems like an extremely slippery slope

**Comment by shortorian** [comment]
Is the edited version any better?

**Comment by asottile** [comment]
no

**Comment by shortorian** [comment]
what's the primary concern with the format operation when we know the argument is in the columns?

**Comment by asottile** [comment]
how do you know every caller from now until forever is only going to pass strings that are in the columns as the first argument
